### PR TITLE
Make sure that only `personal_sign` and `personal_ecRecover` are exported

### LINKED ISSUE #834: Limit exposed `personal_*` APIs
# Problem

As a result of https://github.com/status-im/status-go/pull/822, when the les node starts, it exposes a full set of `personal_` APIs. Some of these might be potentially unsafe to expose.

We need to limit the set of APIs exposed by LES to just `personal_sign` and `personal_recover`.

# Acceptance Criteria

- The only APIs that are available in `personal_*` namespace are `personal_sign` and `personal_recover`.
- DApps (like Crytokitties) work when no upstream is selected.

### PR DESCRIPTION
**Pull request by mandrigin**
+ a unit test for that

Closes #834

### CODE REVIEW

**Review by adambabik — APPROVED** [review]
LGTM

We should start moving this kind of test cases from e2e to unit tests.

File: t/e2e/services/personal_api_test.go [hunk 1]
```diff
@@ -48,6 +57,48 @@ type PersonalSignSuite struct {
...
+		Error *rpcError `json:"error"`
+	}
+
+	s.NoError(json.Unmarshal([]byte(result), &response))
+
+	hidden := (response.Error != nil && response.Error.Code == methodNotFoundErrorCode)
+
+	s.Equal(expectExported, !hidden,
+		"method %s should be %s, but it isn't",
+		method, map[bool]string{true: "exported", false: "hidden"}[expectExported])
```
Comment (themue) [inline t/e2e/services/personal_api_test.go:99]: Hehe, tricky output creation. Would have done mapping in an extra step before.

**Review by themue — APPROVED** [review]
LGTM

**Comment by dshulyak** [comment]
@mandrigin what if restricted api will be made public and personal will be removed from api modules set?

File: services/personal/api.go [hunk 2]
```diff
@@ -62,6 +62,16 @@ func (api *PublicAPI) SetRPC(rpcClient *rpc.Client, timeout time.Duration) {
 	api.rpcTimeout = timeout
 }
 
+// Recover is an implementation of `personal_ecRecover` or `web3.personal.recover` API
```
Comment (gravityblast) [inline services/personal/api.go]: I think also the `web3` function is called `ecRecover` instead of just `recover` right (`web3.personal.ecRecover `)?

Comment (mandrigin) [reply services/personal/api.go]: good catch!

**Comment by mandrigin** [comment]
@dshulyak then we will lose other `personal_` for our internal IPC/RPC, I don't think that is a good idea...

**Comment by mandrigin** [comment]
@adambabik makes sense, next time I'll think about moving it. Though it will require some mocks...

**Comment by dshulyak** [comment]
What will be lost if node.attach ignores private flag at all?

**Comment by mandrigin** [comment]
@dshulyak 
> personal will be removed from api modules set

I maybe misunderstood what that phrase means. If we replace `NewPrivateAccountAPI` in `intenal/ethapi/backend.go` with our limited version, no one will have access to other methods, because there is no object that exposes them. Our limited version quite literally has only 2 methods in it's interface.

**Comment by adambabik** [comment]
@dshulyak  

> what if restricted api will be made public and personal will be removed from api modules set?

If something is removed from the `APIModules` configuration, it is still available in the in-proc RPC handler no matter if it's private or public.

Public is useful only if `APIModules` configuration is empty then public APIs are exposed via HTTP interface.

Source:
* Creating in-proc handler: https://github.com/ethereum/go-ethereum/blob/v1.8.4/node/node.go#L283-L294
* Filtering APIs for HTTP handler: https://github.com/ethereum/go-ethereum/blob/v1.8.4/rpc/endpoints.go#L33-L40 (the same pattern is followed for our `CallRPC` binding now) 

In our case, `CallPrivateRPC` binding should be possible to call any `personal_` methods while `CallRPC` binding should be able to only call two handlers we exposed.

<blockquote><div> GitHub</div><div><strong><a href="https://github.com/ethereum/go-ethereum">ethereum/go-ethereum</a></strong></div><div>go-ethereum - Official Go implementation of the Ethereum protocol</div></blockquote>
<blockquote><div> GitHub</div><div><strong><a href="https://github.com/ethereum/go-ethereum">ethereum/go-ethereum</a></strong></div><div>go-ethereum - Official Go implementation of the Ethereum protocol</div></blockquote>

**Comment by dshulyak** [comment]
This is exactly what i meant. Make sign and ecrecover public, and all other
methods private. Then you remove personal from modules and we will get what
is needed. No?

**Comment by mandrigin** [comment]
@dshulyak that's what I've started from :)
It doesn't work, because of this code checking if you have **anything** in your whitelist.

```go
whitelist[api.Namespace] || (len(whitelist) == 0 && api.Public)
```

It uses `Public` flag only if there is nothing in the whitelist. And we have `eth,net,web3,shh,shhext` in it, so it always ignores this flag.

**Comment by dshulyak** [comment]
thanks, now i see that @adambabik mentioned the same
wasn't good idea to review from phone..

**Review by dshulyak — APPROVED** [review]

**Pull request merged**